src/core_modules/capture-core/components/Pages/common/TEIRelationshipsWidget/useInheritedAttributeValues.js

```diff
@@ -10,17 +10,15 @@ type Props = {
 
 type Return = {
     inheritedAttributes: Array<InputAttribute>,
-    originTeiId: ?string,
     isLoading: boolean,
 };
 export const useInheritedAttributeValues = ({ teiId, trackedEntityTypeId }: Props): Return => {
     const trackedEntityType = getTrackedEntityTypeThrowIfNotFound(trackedEntityTypeId);
-    const inheritedAttributeIds = trackedEntityType.attributes.reduce((acc, attribute) => {
-        if (attribute.inherit) {
-            acc.push(attribute.id);
-        }
-        return acc;
-    }, []);
+    const inheritedAttributeIds = new Set<string>(
+        trackedEntityType.attributes
+            .filter(attribute => attribute.inherit)
+            .map(attribute => attribute.id),
+    );
```

Comment:
Hmm, two passes using `filter` and `map`, and then surely there must be a third one in the `Set` constructor.. I think we should keep reduce but change the initial value from `[]` to `new Set` :grin: